src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1505 @@ const ActorInit En_Elf_InitVars = {
+
+        if (thisx->textId == ElfMessage_GetCUpText(globalCtx)) {
+            this->fairyFlags |= 0x80;
+            gSaveContext.naviTimer = 3001;
+        }
+
+        this->fairyFlags |= 0x10;
+        this->fairyFlags |= 0x20;
+        thisx->update = func_80A052F4;
+        func_80A01C38(this, 3);
+
+        if (this->elfMsg != NULL) {
+            this->elfMsg->actor.flags |= 0x100;
+        }
+
+        thisx->flags &= ~0x10000;
+    } else {
+        this->actionFunc(this, globalCtx);
+        thisx->shape.rot.y = this->unk_2BC;
+        nREG(80) = gSaveContext.sceneFlags[127].chest;
+
+        if (nREG(81) != 0) {
+            if (gSaveContext.sceneFlags[127].chest) {
+                LOG_NUM("z_common_data.memory.information.room_inf[127][ 0 ]", gSaveContext.sceneFlags[127].chest,
+                        "../z_en_elf.c", 2595);
+            }
+        }
+
+        if (!Gameplay_InCsMode(globalCtx)) {
+            if (gSaveContext.naviTimer < 25800) {
+                gSaveContext.naviTimer++;
+            } else if (!(this->fairyFlags & 0x80)) {
+                gSaveContext.naviTimer = 0;
+            }
+        }
+    }
+
+    this->elfMsg = NULL;
+    this->timer++;
+
+    if (this->unk_2A4 > 0.0f) {
+        Math_ApproxF(&this->unk_2A4, 0.0f, 0.05f);
+        func_800773A8(globalCtx, SQ(this->unk_2A4) * this->unk_2A4, player->actor.projectedPos.z + 780.0f, 0.2f, 0.5f);
+    }
+
+    // temp probably fake match
+    unk2C7 = this->unk_2C7;
+    if (unk2C7 > 0) {
+        this->unk_2C7--;
+    }
+
+    if ((this->unk_2C7 == 0) && (globalCtx->csCtx.state != 0)) {
+        this->unk_2C7 = 1;
+    }
+
+    func_80A04D90(this, globalCtx);
+}
+
+void EnElf_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnElf* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    this->actor.shape.rot.y = this->unk_2BC;
+    this->timer++;
+
+    if (this->fairyFlags & FAIRY_FLAG_BIG) {
+        func_80A04D90(this, globalCtx);
+    }
+}
+
+s32 EnElf_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                           struct Actor* thisx, Gfx** gfx) {
+    static Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 pad;
+    f32 scale;
+    Vec3f mtxMult;
+    EnElf* this = THIS;
+
+    if (limbIndex == 8) {
+        scale = ((Math_Sins(this->timer * 4096) * 0.1f) + 1.0f) * 0.012f;
+
+        if (this->fairyFlags & FAIRY_FLAG_BIG) {
+            scale *= 2.0f;
+        }
+
+        scale *= (this->actor.scale.x * 124.99999f);
+        Matrix_MultVec3f(&zeroVec, &mtxMult);
+        Matrix_Translate(mtxMult.x, mtxMult.y, mtxMult.z, MTXMODE_NEW);
+        Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    }
+
+    // do not draw wings for big fairies
+    if (this->fairyFlags & FAIRY_FLAG_BIG) {
+        if (limbIndex == 4 || limbIndex == 7 || limbIndex == 11 || limbIndex == 14) {
+            *dList = NULL;
+        }
+    }
+
+    return false;
```

Comment:
relevant discussion in the [ovl_En_Daiku PR](https://github.com/zeldaret/oot/pull/422#discussion_r499186392)

Reply:
aight yeah I agree with what youre saying there that it should all be done at once.
there was some conversation in discord about it that I think you might have missed, if you wanna scroll back to last night. dragorn and krim said a few words about it

Reply:
I'm fine with the idea if we also change the skelanime side to not just do `== 0` (since that's not treating it as a boolean imo).

But I'm still worried we'll run into a function that returns something other than 1, which will work just fine since the skelanime engine expects anything other than 0, but won't be consistent if we start using true/false.
I'd say to wait and change it once we're sure, but I won't argue further if we're willing to take the chance.

Reply:
I have no issues with that approach. Ill post about it in discord later since the others prob wont see it buried in this pr